specification/dartLangSpec.tex

```diff
@@ -16276,38 +16279,29 @@ \subsection{Assignable Expressions}
 \end{grammar}
 
 \LMHash{}%
-An \Index{assignable expression} is either:
-\begin{itemize}
-\item An identifier.
-\item An invocation (possibly conditional) of a getter (\ref{getters})
-  or list access operator on an expression $e$.
-\item An invocation of a getter or list access operator on \SUPER.
-\end{itemize}
-
-\LMHash{}%
-An assignable expression of the form \id{} is evaluated as
-an identifier expression (\ref{identifierReference}).
+\BlindDefineSymbol{e, t}%
+Let $e$ be an \synt{assignableExpression}.
+Let $t$ be a term such that $e$ can be derived from
+\syntax{$t$ <assignableSelector>}.
+We say that $t$ is the
+\IndexCustom{receiver expression}{assignable expression!receiver expression}
+of the assignable expression $e$.
 
-%An assignable expression of the form \code{$e$.\id($a_1, \ldots,\ a_n$)}
-% is evaluated as a method invocation (\ref{methodInvocation}).
-
-\LMHash{}%
-An assignable expression of the form \code{$e$.\id} or \code{$e$?.\id}
-is evaluated as a property extraction (\ref{propertyExtraction}).
-
-\LMHash{}%
-An assignable expression of the form \code{$e_1$[$e_2$]}
-is evaluated as a method invocation of
-the operator method \code{[]} on $e_1$ with argument $e_2$.
-
-\LMHash{}%
-An assignable expression of the form \code{\SUPER.\id} is evaluated as
-a property extraction.
+\commentary{%
+In short, we obtain $t$ by cutting off an assignable selector
+from the end of $e$.
+It is easy to see that every \synt{assignableExpression}
+except the ones that are simply an \synt{identifier}
+ends in an \synt{assignableSelector}
+(an \synt{unconditionalAssignableSelector}
+is also an \synt{assignableSelector}).
+Note that $t$ is an \synt{expression}:
+Every possible value for $t$ can also be derived from \synt{expression}.%
```

Comment:
Except if `t` is `super`.
And if `t` is an identifier denoting an import prefix, then it's *technically* an expression, because it's an identifier, but not one which can be *evaluated*.
And if `t` denotes a class/mixin/extension declaration, it's also not evaluated (unless we still define static lookup as members on the `Type` object.)

Reply:
Yes, we certainly need to mention those situations where `t` isn't an expression. Done.

Reply:
I think it's considerably more correct now! ;-)